src/Plugin.php

```diff
@@ -77,7 +77,10 @@ public static function request(array $query_vars) {
     if (isset($query_vars['product_cat']) && $query_vars['product_cat'] !== '' && !term_exists($query_vars['product_cat'], 'product_cat')) {
       // If the requested path is a child page of the shop page then query that
       // page instead of a category or product.
-      $pagename = static::getCategoryBase() . '/' . ltrim($query_vars['product_cat_and_post_name'], '/');
+      $pagename = static::getCategoryBase() . '/';
+      if (isset($query_vars['product_cat_and_post_name'])) {
+        $pagename .= ltrim($query_vars['product_cat_and_post_name'], '/');
```

Comment:
The pagename should not contain a trailing slash if no child is appended.
```suggestion
      $pagename = static::getCategoryBase();
      if (isset($query_vars['product_cat_and_post_name'])) {
        $pagename .= '/' . ltrim($query_vars['product_cat_and_post_name'], '/');
```